### ISSUE TITLE
Two relates group generates warning: found multiple \relates, \relatesalso or \memberof commands in a comment block, using last definition

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10835]
**Describe the bug**
I think [the change made to](https://github.com/doxygen/doxygen/issues/851) has introduced the warning  `warning: found multiple \relates, \relatesalso or \memberof commands in a comment block, using last definition`. This showed up with Doxygen 1.8.15 and continues to occur on trunk (eb110639dcc32804b59abdf0dcb343abde211b1e*). The resulting page seems to be correct and the 4 operators were documented.

```
/**
 * Test docs
 */
class Test
{
};

/**
 * \relates Test
 *
 * operator== docs
 */
///@{
operator==(Test&, Test&);

operator==(Test&, int);
///@}

/**
 * \relates Test
 *
 * operator!= docs
 */
///@{
operator!=(Test&, Test&);

operator!=(Test&, int);
///@}
```

**To Reproduce**
[testcase.zip](https://github.com/doxygen/doxygen/files/15153071/testcase.zip)

**Expected behavior**
No warning produced.

**Version**
Doxygen 1.8.15 - trunk (eb110639dcc32804b59abdf0dcb343abde211b1e*)

**Comment by scott-zhong** [comment]
Using `\relates` on grouping prior to other `\relates` documentation also reproduces the warning using Doxygen 1.11.0
```
/**
 * Base
 **/
class Base {
};

/**
 * \relates Base
 * fun
 */
///@{
void b();
void a();
int CC;
int CA;
int CB;
///@}

/**
 * \relates Base
 *
 * fun
 */
void d();

/**
 * \relates Base
 *
 * fun
 */
void c();
```

[testcase.zip](https://github.com/user-attachments/files/16089326/testcase.zip)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10985

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
